src/components/withViewportOffsetTop.tsx

```diff
@@ -0,0 +1,41 @@
+import React, {useEffect, forwardRef, useState, ComponentType, RefAttributes, ForwardedRef} from 'react';
+import getComponentDisplayName from '../libs/getComponentDisplayName';
+import addViewportResizeListener from '../libs/VisualViewport';
+
+type ViewportOffsetTopProps = {
+    // viewportOffsetTop returns the offset of the top edge of the visual viewport from the
+    // top edge of the layout viewport in CSS pixels, when the visual viewport is resized.
+    viewportOffsetTop: number;
+};
+
+export default function withViewportOffsetTop<TProps extends ViewportOffsetTopProps, TRef>(WrappedComponent: ComponentType<TProps & RefAttributes<TRef>>) {
+    function WithViewportOffsetTop(props: Omit<TProps, keyof ViewportOffsetTopProps>, ref: ForwardedRef<TRef>) {
+        const [viewportOffsetTop, setViewportOffsetTop] = useState(0);
+
+        useEffect(() => {
+            const updateDimensions = (e: Event) => {
+                const targetOffsetTop = (e.target as HTMLElement)?.offsetTop ?? 0;
```

Comment:
Instead of assertion:
```suggestion
                const targetOffsetTop = (event.target instanceof HTMLElement && event.target.offsetTop) || 0;
```

Please test before commiting

Reply:
updated and tested